topsort-api-v2.yml

```diff
@@ -466,6 +515,230 @@ components:
+        id:
+          type: string
+          description: >
+            The marketplace's unique ID for the click.
+            This field ensures the event reporting is idempotent in case there is a network issue and the request is retried.
+            If there is no click model on the marketplace side, generate a unique string that does not change if the event is resent.
+          example: b39d39ed-ea0e-4059-9d15-4990b39c85a2
+    Purchase:
+      description: >
+        A purchase is sent to Topsort once the user makes a payment transaction and it has been confirmed by the payment provider or financial entity in charge processing the transaction.
+      type: object
+      required:
+        - occurredAt
+        - opaqueUserId
+        - items
+        - id
+      properties:
+        occurredAt:
+          type: string
+          description: RFC3339 formatted timestamp including UTC offset
+          example: '2021-10-12T07:20:50.52Z'
+          format: date-time
+        opaqueUserId:
+          type: string
+          description: >
+            The opaque user ID allows correlating user activity, whether or not they are actually logged in.
+            It must be long lived.
+          example: 71303ce0-de89-496d-8270-6434589615e8
+        items:
+          type: array
+          description: Items purchased.
+          minItems: 1
+          items:
+            $ref: '#/components/schemas/PurchaseItem'
+        id:
+          type: string
+          description: >
+            The marketplace unique ID for the order.
+            This field ensures the event reporting is idempotent in case there is a network issue and the request is retried.
+            If there is no unique ID for orders on the marketplace side, generate a unique string that does not change if the event is resent.
+          example: 0e06c899-b2cd-4e0d-b0de-8aefb4b6d0a0
+    PurchaseItem:
+      type: object
+      required:
+        - productId
+        - unitPrice
+      properties:
+        productId:
+          type: string
+          description: The marketplace ID of the product being purchased.
+          example: p_SA0238
+        quantity:
+          type: integer
+          minimum: 1
+          default: 1
+          description: Count of products purchased.
+          example: 2
+        unitPrice:
+          type: number
+          format: float
+          minimum: 0.0
+          exclusiveMinimum: true
+          description: The price of a single item in the marketplace currency.
+          example: 12.95
+    EventsResponse:
+      type: object
+      properties:
+        impressions:
+          type: object
+          required:
+            - ids
+          properties:
+            ids:
+              type: array
+              description: An array of impression IDs that were reported.
+              items:
+                type: string
+                description: The marketplace's ID for the impression.
```

Comment:
Why do we need to return the id that was passed in the request?

Reply:
Agreed, we could just return a 204.